@RolesAllowed() not working

I have a HTTP webservice providing REST and SOAP endpoints:

Here's the REST service:
```
@RequestScoped
public class FruitResource {

    @GET
    @Path("/list")
    @RolesAllowed("USER")
    public List<String> list() {
         final SecurityIdentity securityIdentity =CDI.current()
                 .select(SecurityIdentity.class)
                 .get();
        System.out.println(securityIdentity.hasRole("USER"));
        return Arrays.asList( .. );
    }
}
```

Here's the "same" service as SOAP:

```
@WebService(endpointInterface = "org.acme.cxf.FruitWebService")
public class FruitWebServiceImpl implements FruitWebService {

    @Override
    @RolesAllowed("USER")
    public List<String> list() {
        final SecurityIdentity securityIdentity =CDI.current()
                 .select(SecurityIdentity.class)
                 .get();
        System.out.println(securityIdentity.hasRole("USER"));
        return Arrays.asList( .. );
    }
```

This does not work at all.  I'm getting a `javax.enterprise.context.ContextNotActiveException: null` in `io.quarkus.arc.impl.ClientProxies#getDelegate` while all works fine using the REST EP.

My configuration is rather simple:
```
quarkus.http.auth.basic=true
quarkus.security.users.embedded.enabled=true
quarkus.security.users.embedded.plain-text=true
quarkus.security.users.embedded.users.x=x
quarkus.security.users.embedded.roles.x=USER     
quarkus.resteasy.path=/rest
quarkus.cxf.path=/soap
quarkus.cxf.endpoint."/fruit".implementor=org.acme.cxf.FruitWebServiceImpl
```

Not sure whether this is a bug. However, I somehow expect that configured (http) security works regardless of the service implementation.

In addition I created a variation of this issue (--> branch: https://github.com/wh81752/cxfdemo-quarkus/tree/service-rbac). 

Again a HelloWorld REST Resource and SOAP Endpoint. Security / RBAC (Role Based Access) is applied to a shared service implementation --> https://github.com/wh81752/cxfdemo-quarkus/blob/service-rbac/src/main/java/io/app/rest/HelloWorldResource.java

Accessing the service via REST works as expected:
```
curl -u u:p -v http://localhost:8080/rest/hw/hello         ;;; 200
curl            -v http://localhost:8080/rest/hw/hello         ;;; 401
```

Accessing `hello()` via SOAP/CXF produces:
```
Caused by: javax.enterprise.context.ContextNotActiveException
        at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:40)
        at io.quarkus.security.runtime.SecurityIdentityProxy_ClientProxy.arc$delegate(SecurityIdentityProxy_ClientProxy.zig:42)
        at io.quarkus.security.runtime.SecurityIdentityProxy_ClientProxy.hasRole(SecurityIdentityProxy_ClientProxy.zig:398)
        at io.quarkus.security.runtime.interceptor.check.RolesAllowedCheck.apply(RolesAllowedCheck.java:54)
        at io.quarkus.security.runtime.interceptor.SecurityConstrainer.check(SecurityConstrainer.java:27)
        at io.quarkus.security.runtime.interceptor.SecurityHandler.handle(SecurityHandler.java:23)
        at io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor.intercept(RolesAllowedInterceptor.java:29)
        at io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor_Bean.intercept(RolesAllowedInterceptor_Bean.zig:386)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at io.app.service.HelloWorldService_Subclass.hello(HelloWorldService_Subclass.zig:568)
        at io.app.soap.HelloWorldImpl.hello(HelloWorldImpl.java:26)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179)
        at org.apache.cxf.jaxws.JAXWSMethodInvoker.performInvocation(JAXWSMethodInvoker.java:66)
        at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96)
```

--- a/integration-tests/src/main/java/io/quarkiverse/it/cxf/GreetingWebServiceImpl.java
+++ b/integration-tests/src/main/java/io/quarkiverse/it/cxf/GreetingWebServiceImpl.java
@@ -1,5 +1,6 @@
 package io.quarkiverse.it.cxf;
 
+import javax.annotation.security.RolesAllowed;
 import javax.inject.Inject;
 import javax.jws.WebParam;
 import javax.jws.WebService;

--- a/runtime/src/main/java/io/quarkiverse/cxf/transport/VertxHttpServletRequest.java
+++ b/runtime/src/main/java/io/quarkiverse/cxf/transport/VertxHttpServletRequest.java
@@ -36,7 +36,10 @@ import javax.servlet.http.Part;
 import org.apache.cxf.common.util.UrlUtils;
 import org.jboss.logging.Logger;
 
+import io.quarkus.security.identity.CurrentIdentityAssociation;
+import io.quarkus.security.identity.SecurityIdentity;
 import io.quarkus.vertx.http.runtime.VertxInputStream;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
 import io.vertx.core.http.HttpServerRequest;
 import io.vertx.core.http.HttpServerResponse;
 import io.vertx.ext.web.RoutingContext;
@@ -289,7 +292,7 @@ public class VertxHttpServletRequest implements HttpServletRequest {
     @Override
     public boolean isSecure() {
         LOG.trace("isSecure");
-        return false;
+        return request.isSSL();
     }
 
     @Override
@@ -330,8 +333,12 @@ public class VertxHttpServletRequest implements HttpServletRequest {
 
     @Override
     public String getAuthType() {
-        LOG.trace("getAuthType");
-        return "null";
+        String authorizationValue = request.getHeader("Authorization");
+        if (authorizationValue == null) {
+            return null;
+        } else {
+            return authorizationValue.split(" ")[0].trim();
+        }
     }
 
     @Override
@@ -464,8 +471,11 @@ public class VertxHttpServletRequest implements HttpServletRequest {
 
     @Override
     public Principal getUserPrincipal() {
-        LOG.trace("getUserPrincipal");
-        return null;
+        QuarkusHttpUser user = (QuarkusHttpUser) context.user();
+        if (user == null || user.getSecurityIdentity().isAnonymous()) {
+            return null;
+        }
+        return user.getSecurityIdentity().getPrincipal();
     }
 
     @Override
@@ -495,8 +505,11 @@ public class VertxHttpServletRequest implements HttpServletRequest {
 
     @Override
     public boolean isUserInRole(String role) {
-        LOG.trace("isUserInRole");
-        return false;
+        SecurityIdentity user = CurrentIdentityAssociation.current();
+        if (role.equals("**")) {
+            return !user.isAnonymous();
+        }
+        return user.hasRole(role);
     }
 
     @Override

